### PR TITLE
Improved memory management in the following BBF functions 

### PR DESCRIPTION
**Pull request by lejaokri**
I started running JDBC test with valgrind enabled. So far fixed a couple of minor and major issues.
I'd like to deliver some of them now. More information in Jira

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by surendravishnoi — APPROVED** [review]

**Review by timchang514 — APPROVED** [review]

File: contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c [hunk 1]
```diff
@@ -3185,6 +3187,7 @@ TdsSendTypeNumeric(FmgrInfo *finfo, Datum value, void *vMetaData)
 		decString[precision++] = '0';
 	}
 	decString[precision] = '\0';
+	Assert(precision <= outlen);
```
Comment (Deepesh125) [inline contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c:3190]: wouldn't be `precision == outlen` be better?

File: contrib/babelfishpg_tsql/src/pl_comp.c [hunk 2]
```diff
@@ -1605,9 +1604,7 @@ pltsql_post_expand_star(ParseState *pstate, ColumnRef *cref, List *l)
 				/*
 				 * We found the original name; rewrite it as bbf_original_name
 				 */
-				bbf_original_name = &optstr[18];
-				bbf_original_name[strlen(te->resname)] = '\0';
-				te->resname = pstrdup(bbf_original_name);
+				te->resname = pnstrdup((char *) &optstr[18], strlen(te->resname));
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/pl_comp.c:1607]: Just trying to understand: why would valgrind complain here?

**Review by Deepesh125 — APPROVED** [review]

File: contrib/babelfishpg_tsql/src/pl_comp.c (hunk 2 above)
Comment (lejaokri) [reply contrib/babelfishpg_tsql/src/pl_comp.c:1607]: It's writing one byte passed the length. If bbf_original_name = palloc(10), then line 1609 does a bbf_original_name[10] = '0';

File: contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c (hunk 1 above)
Comment (lejaokri) [reply contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c:3190]: I will take a look deeper

Comment (lejaokri) [reply contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c:3190]: Yes you're right. I will make this change in the next batch of fixes

**Pull request merged**